dev/devicelab/lib/microbenchmarks.dart

```diff
@@ -64,6 +64,11 @@ Future<Map<String, double>> readJsonResults(Process process) {
       // See https://github.com/flutter/flutter/issues/19208
       process.stdin.write('q');
       await process.stdin.flush();
+
+      // Give the process a couple of seconds to exit and run shutdown hooks
+      // before sending kill signal.
+      await Future<void>.delayed(const Duration(seconds: 2));
```

Comment:
has this been an issue, or are you adding this just in case?

Reply:
actually, reading the comment from lines 61-64, I actually fixed that in https://github.com/flutter/flutter/pull/99871.

Were there particular shutdown hooks you were concerned about?

Reply:
It was an issue when I ran the test locally because the shutdown hook which kills Xcode wasn't being triggered because the kill signal was sent before the `q` input could even be processed. 

I don't know if it's been an issue in the past

Reply:
This is the shutdown hook I was having issues with: https://github.com/vashworth/flutter/blob/master/packages/flutter_tools/lib/src/ios/devices.dart#L868-L871

Reply:
so actually, do we run the shutdownHooks when we send `q`? Let me check.

Reply:
Yes, shutdown hooks are run when we send `q`. 

It seems like shutdown hooks aren't run when you send a sigint kill signal

Reply:
well, this would be a serious bug, but anyway, this is fine as is for now, i'll file a tracking issue for us not handling SIGINT correctly.